pkg/chain/ethereum/keep_mode.go

```diff
@@ -6,30 +6,62 @@ import (
 	"math/big"
 	"time"
 
-	"github.com/ethereum/go-ethereum/common"
+	"github.com/keep-network/keep-core/pkg/chain"
 
-	"github.com/ipfs/go-log"
+	"github.com/ethereum/go-ethereum/common"
 
+	"github.com/keep-network/keep-common/pkg/chain/ethereum"
 	"github.com/keep-network/keep-common/pkg/chain/ethereum/ethutil"
 	"github.com/keep-network/keep-common/pkg/subscription"
-	"github.com/keep-network/keep-core/pkg/chain"
 	eth "github.com/keep-network/keep-ecdsa/pkg/chain"
 	"github.com/keep-network/keep-ecdsa/pkg/chain/gen/contract"
 	"github.com/keep-network/keep-ecdsa/pkg/ecdsa"
 	"github.com/keep-network/keep-ecdsa/pkg/utils/byteutils"
 )
 
-var logger = log.Logger("keep-chain-eth-ethereum")
+type KeepStakeChainHandle struct {
```

Comment:
Not sure about this change. We would duplicate most of the `EthereumChain` for two keep types. Please see my previous comment about adding `fullyBackedBondedECDSAKeepFactoryContract` to `EthereumChain`.